labman/gui/static/js/plateViewer.js

```diff
@@ -536,13 +544,9 @@ function SampleCellEditor(args) {
   // to choose the sample from the autocomplete dropdown menu
   this.keyCaptureList = [Slick.keyCode.UP, Slick.keyCode.DOWN];
 
-  this.blanknames = [];
-  $.ajax({
-    url: '/sample/control?term=',
-    success: function (result) {
-      that.blankNames = result;
-    }
-  });
+  // Get the parsed values from the request. Alternatively, if the request
+  // didn't complete yet use an empty list as a fallback.
```

Comment:
Is it a failure if the request does not complete?

Reply:
I think so, yes. Is it possible to block until the ajax completes followed by a test of a status code?

Reply:
Blocking requests are frowned upon and deprecated (or on their way out). This [link](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests) has information (see under Synchronous Requests).

For now, I'll add some error handling and let the user know if they need to reload the page to try again.

Reply:
I've added some error handling (a message at the top of the screen describing the problem and a possible workaround). Note that for this test I modified the URI so the request pointed to a non-existant route hence the 404.

![screen shot 2018-07-25 at 12 50 27 pm](https://user-images.githubusercontent.com/375307/43223908-77c9758e-9009-11e8-8c8c-e6785374ad7f.png)

Reply:
:+1: